File: gb-spring-backend/src/main/java/com/egroupbuy/model/OrderDetail.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class OrderDetail {
    int order_details_id;
    int order_id;
    int product_id;
    int seller_id;
    String seller_username;
    String product_title;
    double product_unit_price;
    String product_image_url;
    String status;
    int quantity;
    double sub_total;
    Timestamp delivery_date;
    String sell_type;
    int package_grp_id;
    String package_grp_name;
    int package_id;
    String package_name;
    Timestamp grpbuy_end;
    int grpbuy_allocate_qty;
    String grpbuy_result;
    String shipping_method;
    Timestamp order_date;
    List<PackageItem> package_item_list;
}

File: gb-angular-frontend/src/app/interfaces/order-detail.ts
import {PackageItem} from './package-item';

export interface OrderDetail {
  order_details_id?: number;
  order_id?: number;
  product_id: number;
  seller_id: number;
  seller_username: string;
  product_title: string;
  product_unit_price: number;
  product_image_url: string;
  status: string;
  quantity: number;
  sub_total: number;
  delivery_date: string;
  sell_type: string;
  package_grp_id?: number;
  package_grp_name?: string;
  package_id?: number;
  package_name?: string;
  grpbuy_end?: string;
  grpbuy_allocate_qty?: number;
  grpbuy_result?: string;
  shipping_method?: string;
  order_date?: string;
  package_item_list?: PackageItem[];
}
